File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    var numsSet = new Set(nums);
    var longestStreak = 0;
    
    for (num of numsSet) {
        if (!(numsSet.has(num-1))) {
            currNum = num;
            currStreak = 1;
            
            while (numsSet.has(currNum+1)) {
                currNum++;
                currStreak++;
            }
            
            longestStreak = Math.max(currStreak, longestStreak);
        }
    }
    
    return longestStreak;
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) is 0:
            return 0
        numSet = set(nums)
        maxLen = 0
        
        for currVal in numSet:
            if currVal-1 not in numSet:
                currSeqLen = 0
                while currVal+currSeqLen in numSet:
                    currSeqLen += 1
                maxLen = max(maxLen, currSeqLen)

        return maxLen